src/uu/sort/src/sort.rs

```diff
@@ -1158,12 +1179,43 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         .map(String::from);
 
     if let Some(n_merge) = matches.get_one::<String>(options::BATCH_SIZE) {
-        settings.merge_batch_size = n_merge.parse().map_err(|_| {
-            UUsageError::new(
-                2,
-                format!("invalid --batch-size argument {}", n_merge.quote()),
-            )
-        })?;
+        match n_merge.parse::<usize>() {
+            Ok(parsed_value) => {
+                if parsed_value < 2 {
+                    show_error!("invalid --batch-size argument '{}'", n_merge);
+                    return Err(UUsageError::new(2, "minimum --batch-size argument is '2'"));
+                }
+                settings.merge_batch_size = parsed_value;
+            }
+            Err(e) => {
+                if *e.kind() == std::num::IntErrorKind::PosOverflow {
+                    #[cfg(target_os = "linux")]
+                    {
+                        show_error!("--batch-size argument {} too large", n_merge.quote());
+
+                        return Err(UUsageError::new(
+                            2,
+                            format!(
+                                "maximum --batch-size argument with current rlimit is {}",
+                                get_rlimit()
+                            ),
+                        ));
+                    }
+                    #[cfg(not(target_os = "linux"))]
+                    {
+                        return Err(UUsageError::new(
+                            2,
+                            format!("--batch-size argument {} too large", n_merge.quote()),
+                        ));
+                    }
+                } else {
+                    return Err(UUsageError::new(
+                        2,
+                        format!("invalid --batch-size argument {}", n_merge.quote()),
+                    ));
+                }
```

Comment:
An idea: to move the `return Err` outside of the condition block and make the condition return an error message.